NewForce-at-Mountwest/bangazonapi-team-horrohawk
# Product - Ticket #2 - CRUD - PR Attempt #1.

This feature introduces Products to the Bangazon Inc e-commerce business. Ticket 2 completes all endpoints to GET, GET single, POST, PUT, and DELETE to the Product table/list. These additions also allow for finding products by type and by customer. 

## Related Issues
This ticket will function independently from other tickets once you have run the SQL script to build the database and seed data using the file "bangazonapi-team-horrohawk/TablesAndSeedData" from our code repository.

## Steps To Test
clone down the project to your computer
cd into BangazonAPI
from command line, type in "start BangazonAPI.sln" then press Enter to open in Visual Studio
Create the database by opening the 'SQL Server Object Explorer'
Right click [MACHINE NAME]/SQLEXPRESS select 'New Query'
Copy the first script in 'TablesAndSeedData' then paste it and click the green play button top left
Repeat the same steps for the second script in the file to seed the database with data
Click the other green play button in the top of the page
Then open Postman and select GET and type 'http://localhost:63540/api/product' in the text field to get all products in the database
Then in Postman select GET and type 'http://localhost:63540/api/product/[id]' in the text field to get a single product in the database
Then in Postman select POST and type 'http://localhost:63540/api/product' in the text field and in the BODY, copy this example, then paste it and click send
{
        
        "price": 3.99,
        "title": "Wilson Basketball",
        "description": "Indoor Basketball for winter time.",
        "quantity": 1,
        "productTypeId": 1,
        "customerId": 1
  
    }
Then use that same object, just edit the description.
Select PUT and click send, that should then edit the current object with that Id.
Finally select DELETE and type 'http://localhost:63540/api/product/[id of the created object]' and click send. It should remove that object from the product database.
It should only allow you to delete the product that the user created.
Close out Postman if everything is functioning.

## What kind of change does this PR introduce?
The change is adding the Product to Bangazon Inc for team-horrohawk
Please check one of the following:
